zulipterminal/ui_tools/views.py

```diff
@@ -880,13 +880,13 @@ def keypress(self, size: urwid_Size, key: str) -> str:
 class StreamInfoView(PopUpView):
     def __init__(self, controller: Any, color: str,
                  name: str, desc: str) -> None:
-        # TODO: Height handling could be improved
         # Add 4 (for 2 Unicode characters on either side) to the popup title
         # length to make sure that the title gets displayed even when the
         # content is shorter than the title length (+4 Unicode characters).
         self.width = max(len(desc) + 2, len("# {}".format(name)) + 4)
-        self.height = 2
-        stream_info_content = [urwid.Text(desc, align='center')]
+        description = urwid.Text(desc, align='center')
+        self.height = description.rows((self.width, ))
```

Comment:
I was just going to merge this, but realized we have a subtle nuance here, as `self.width` is always requested to be big enough to hold the description, so we end up wit ha height of 1. The popup can be scrolled, but that's probably not what we want!

Given that the height is only known when showing the popup, we likely want to use a different approach for cases where we want some kind of minimum height visible, or use the column width we know we get in the popup to calculate it. The controller popup code needs improving, whether a special case for a 'minimum height visible' case like this, or a general change where the height is calculated later, maybe all from rows?

As part of this, we may wish to limit the maximum popup width, like we do with popup height?